server/src/main/java/org/elasticsearch/common/lucene/search/function/ScriptScoreQuery.java

```diff
@@ -143,7 +148,12 @@ public Explanation explain(LeafReaderContext context, int doc) throws IOExceptio
                         explanation = Explanation.match(score, desc);
                     }
                 }
-                
+                if (boost != 1.0f) {
```

Comment:
This seems to be executed also when `explanation == null`? Or I'm missing something?
Maybe it worths adding a test where the boost is ` != 1` and check that there is no explanation returned regarding the boost?

Also, why not return an explanation if the boost is `1`, since it's asked by the user?

Reply:
The way we handle `explanation` object is different from other queries, which may be confusing. In `script_score` query a user can provide his/her custom explanation. 

> This seems to be executed also when explanation == null?

Not exactly. We start with `explanation` that is user provided explanation for the script, and if it null, it is substituted by our standard explanation (line 143), so on in line 151 explanation is never null. 

> Maybe it worths checking for a test that when the boost is != 1 and explanation is false there is no explanation returned regarding the boost?

When is explanation is not asked (`explain = false` in a search request),  we will not even go to this method. This method is executed only when a user requests an explanation. 

> Also, why not return an explanation if the boost is 1, since it's asked by the user?

I was thinking since `boost` is an optional parameter,  when it is not provided by a user, there is no need to provide an explanation about it.